protocols/kad/src/behaviour/test.rs

```diff
@@ -1317,3 +1317,67 @@ fn network_behaviour_inject_address_change() {
         kademlia.addresses_of_peer(&remote_peer_id),
     );
 }
+
+#[test]
+fn client_mode() {
+    let mut cfg = KademliaConfig::default();
+    cfg.set_mode(Mode::Client);
+
+    // Create a server and client peer.
+    let client_swarm = build_node_with_config(cfg);
+    let server_swarm = build_node_with_config(KademliaConfig::default());
+
+    let mut swarms = vec![client_swarm, server_swarm];
+
+    // Collect addresses and peer IDs.
+    let addrs: Vec<_> = swarms.iter().map(|(addr, _)| addr.clone()).collect();
+    let peers: Vec<_> = swarms
+        .iter()
+        .map(|(_, swarm)| swarm.local_peer_id().clone())
+        .collect();
+
+    swarms[0].1.dial_addr(addrs[1].clone()).unwrap();
+    swarms[1].1.dial_addr(addrs[0].clone()).unwrap();
+
+    block_on(poll_fn(move |ctx| {
+        // Flag variables to store the result of various checks.
+        let mut is_server_present = false;
+        let mut is_client_absent = false;
+
+        for (_, swarm) in swarms.iter_mut() {
+            loop {
+                match swarm.poll_next_unpin(ctx) {
+                    Poll::Ready(Some(SwarmEvent::Behaviour(KademliaEvent::RoutingUpdated {
+                        peer,
+                        ..
+                    }))) => {
+                        // Check if the server peer is present in the client peer's routing table.
+                        if swarm.local_peer_id().clone() == peers[0] {
+                            is_server_present = peer == peers[1];
+                        }
+                        return Poll::Ready(());
+                    }
+                    Poll::Ready(_) => {
+                        // Check if the client peer is NOT present in the server peer's routing table.
+                        if swarm.local_peer_id().clone() == peers[1] {
+                            is_client_absent = swarm.behaviour_mut().kbucket(peers[0]).is_none();
```

Comment:
It is kind of hard to reliably check for absence of something, especially in an async setting.

How about:

1. The server stores a value `Kademlia::put_value`, waiting for a `KademliaEvent::OutboundQueryCompleted`.
2. The client connects to the server.
3. The client does a `Kademlia::get_record`, waiting for a `KademliaEvent::OutboundQueryCompleted`.
4. Check that the client has the server in its routing table and check that the server does not have the client in its routing table.

Reply:
Sure. I will implement this. I will try to finish before the same time, tomorrow. Please keep an eye out for any notifications from this PR. I want to get this PR closed, as early as possible (ofcourse, without sacrificing code quality).